lib/spack/docs/basic_usage.rst

```diff
@@ -284,6 +284,86 @@ and removed everything that is not either:
 You can check :ref:`cmd-spack-find-metadata` to see how to query for explicitly installed packages
 or :ref:`dependency-types` for a more thorough treatment of dependency types.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Marking packages explicit or implicit
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Packages can be marked as explicitly or implicitly installed by using ``spack mark``.
+This can be used in combination with ``spack gc`` to clean up packages that are
+no longer required.
+
+.. code-block:: console
+
+  $ spack find -d
+  ==> 4 installed packages
+  -- linux-fedora32-haswell / gcc@10.1.1 --------------------------
+  libsigsegv@2.11
+
+  libsigsegv@2.12
+
+  m4@1.4.18
+      libsigsegv@2.12
+
+  m4@1.4.18
+      libsigsegv@2.11
+
+  $ spack gc
+  ==> There are no unused specs. Spack's store is clean.
+
+  $ spack mark -i m4 ^libsigsegv@2.11
+  ==> m4@1.4.18 : marking the package implicit
+
+  $ spack gc
+  ==> The following packages will be uninstalled:
+
+      -- linux-fedora32-haswell / gcc@10.1.1 --------------------------
+      5fj7p2o libsigsegv@2.11  c6ensc6 m4@1.4.18
+
+  ==> Do you want to proceed? [y/N]
+
+In the example above, we ended up with two versions of ``m4`` since they depend
+on different versions of ``libsigsegv``. ``spack gc`` will not remove any of
+the packages since both versions of ``m4`` have been installed explicitly.
+
```

Comment:
Why are both versions of `m4` considered being explicitly installed?    Were they installed with `spack install m4 ^libsigsegv@2.11` and ` spack install m4 ^libsigsegv@2.12`? If so, that makes sense to me.

If `spack install m4` results in `libsigsegv@2.12` getting marked as `explicit`, I'm confused.  Are all packages marked as `explict` until someone comes by and manually marks them as `implicit`?  I could be wrong, but I think other package managers like apt use `explicit` to mean the package was explicitly mentioned by the user in a command line invocation, and `implicit` means the package was pulled in as a dependency of an `explicit` package.  If the terms used here differ, it might be worth precisely defining what `explicit` and `implicit` mean in this documentation.

Reply:
> Why are both versions of `m4` considered being explicitly installed? Were they installed with `spack install m4 ^libsigsegv@2.11` and ` spack install m4 ^libsigsegv@2.12`? If so, that makes sense to me.

Exactly. I have now added some more lines to the example that should hopefully make this more clear.

> If `spack install m4` results in `libsigsegv@2.12` getting marked as `explicit`, I'm confused. Are all packages marked as `explict` until someone comes by and manually marks them as `implicit`?

No, only the two `m4` packages will be marked as explicitly installed. I have also added a few sentences explaining this.

Thanks for your feedback!

Reply:
Awesome!  Thanks for the clarifications.  LGTM!